mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/entity/PropertyDefinition.java

```diff
@@ -52,7 +53,8 @@
 
   /**
    * @return The name of the "set" method associated with this property used to update the value of
-   *     the property on the entity.
+   *     the property on the entity, or {@code null} if the entity was marked as {@link
+   *     IntrospectionStrategy#immutable()}.
    */
   String getSetterName();
```

Comment:
Should we convert to `Optional<String>` instead?

Reply:
The method is never called for an immutable entity. I'm not convinced that we should use optional everywhere for internal stuff.